src/frontend/src/app/registration/registration.component.html

```diff
@@ -0,0 +1,68 @@
+<br/>
+<div class="row justify-content-center">
+  <form [formGroup] = "registrationForm" (ngSubmit)="register()">
+
+    <div class="form-group">
+      <label for="inputFirstname">Firstname</label>
+      <input type="text" class="form-control w-50" id="inputFirstname"
+             placeholder="Enter firstname" [(ngModel)]="userDto.firstname" formControlName="firstname">
+      <div *ngIf="registrationForm.controls.firstname.invalid && (registrationForm.controls.firstname.touched || registrationForm.controls.firstname.dirty)">
+      <small id="firstnameError" class="form-text text-danger" *ngIf="registrationForm.controls.firstname.errors.required">firstname is required.</small>
+      </div>
+      </div>
+
+    <div class="form-group">
+      <label for="inputLastname">Lastname</label>
+      <input type="text" class="form-control w-50" id="inputLastname"
+             placeholder="Enter lastname" [(ngModel)]="userDto.lastname" formControlName="lastname">
+      <div *ngIf="registrationForm.controls.lastname.invalid && (registrationForm.controls.lastname.touched || registrationForm.controls.lastname.dirty)">
+      <small id="lastnameError" class="form-text text-danger" *ngIf="registrationForm.controls.lastname.errors.required">lastname is required.</small>
+      </div>
+    </div>
+      <div class="form-group">
+      <label for="inputUsername">Username</label>
+      <input type="text" class="form-control w-50" id="inputUsername" aria-describedby="usernameHelp"
+             placeholder="Enter username" [(ngModel)]="userDto.username" formControlName="username">
+      <small id="usernameHelp" class="form-text text-muted">provide username that will be used to login</small>
+      <div *ngIf="registrationForm.controls.username.invalid && (registrationForm.controls.username.touched || registrationForm.controls.username.dirty)">
```

Comment:
ten if musi się znajdować w kontrolerze czyli w komponencie, stwórz na to jakąś funkcję, a tu tylko ngIf=i nazwa funkcji